src/app/Models/Traits/SpatieTranslatable/HasTranslations.php

```diff
@@ -213,4 +198,38 @@ public function __call($method, $parameters)
                 break;
         }
     }
+
+    /**
+     * @param  \Illuminate\Database\Eloquent\Model  $model
+     * @param  array  $attributes
+     * @param  string  $locale
+     */
+    private static function splitNonTranslatableFromTranslatableAttributes($model, array $attributes, string $locale)
+    {
+        $non_translatable = [];
+        foreach ($attributes as $attribute => $value) {
+            // the attribute is not translatable, store it without aditional process
+            if (! $model->isTranslatableAttribute($attribute)) {
+                $non_translatable[$attribute] = $value;
+                continue;
+            }
+
+            // in case case it's an array, we will check if the keys of the array match the possible translation locales,
+            // if they do, we will set the attribute translations directly from the array.
+            if (is_array($value)) {
+                $possibleTranslations = array_keys($value);
+                $translatableLocales = array_keys($model->getAvailableLocales());
+
+                //if the array keys match the translatable locales (all keys must match some locale, ['en' => something, 'pt' => qualquer])
+                if ($possibleTranslations === array_intersect($possibleTranslations, $translatableLocales)) {
+                    $model->setTranslations($attribute, $value);
+                    continue;
+                }
+            }
+
+            $model->setTranslation($attribute, $locale, $value);
+        }
+
+        return [$model, $non_translatable];
```

Comment:
The signature of this method is a bit odd:
- it returns the $model like it does something on the model;
- but it also returns the non-translatable attributes;

By its name I'd say it returns too things:
- an array of translatable attributes
- an array of non-translatable attributes

And that it doesn't change anything. But that's not the case. It both "does stuff" and "returns stuff". What can we do to improve this @pxpm ?

Reply:
Okay 👍 You rigth!

I splitted into two methods:

`getNonTranslableAttributes` and `fillModelWithTranslations`